### PR TITLE
Fix up print view text for various rental modes

### PR DESCRIPTION
**Pull request by landonreed**
- Properly show custom icons for all modes
- Improve wording for various rental modes
- Add distance traveled to access legs (except when a leg is a hailed car)

Fixes https://github.com/ibi-group/trimet-mod-otp/issues/238

### CODE REVIEW

**Comment by evansiroky** [comment]
Ready for your review, @landonreed.

File: lib/components/narrative/printable/printable-itinerary.js [hunk 1]
```diff
@@ -92,19 +134,49 @@ class AccessLeg extends Component {
...
   render () {
-    const { leg } = this.props
+    const { configCompanies, customIcons, leg } = this.props
+
+    // calculate leg mode label in a special way for this component
+    let legModeLabel = getLegModeLabel(leg)
+
+    if (leg.rentedBike) {
+      // FIXME: Special case for TriMet that needs to be refactored to
+      // incorporate actual company.
```
Comment (evansiroky) [inline lib/components/narrative/printable/printable-itinerary.js]: I dislike writing new FIXMEs, but I figured this could be accomplished while refactoring the OTP vehicle rental updaters.

File: lib/components/app/responsive-webapp.js [hunk 2]
```diff
@@ -193,7 +193,10 @@ class RouterWrapper extends Component {
             />
             <Route
               path='/print'
-              component={PrintLayout}
+              component={(routerProps) => {
+                const props = {...this.props, ...routerProps}
+                return <PrintLayout {...props} />
+              }}
```
Comment (landonreed) [inline lib/components/app/responsive-webapp.js]: Can you please add a comment describing why this change is necessary?

File: lib/components/narrative/printable/printable-itinerary.js [hunk 3]
```diff
@@ -92,19 +134,49 @@ class AccessLeg extends Component {
...
+      )
+      legModeLabel = `Drive ${companiesLabel} ${leg.from.name}`
+    } else if (leg.rentedVehicle) {
+      const companiesLabel = getCompaniesLabelFromNetworks(
+        leg.from.networks,
+        configCompanies
+      )
+      legModeLabel = `Ride ${companiesLabel} eScooter`
+    }
+
```
Comment (landonreed) [inline lib/components/narrative/printable/printable-itinerary.js]: I'm worried that this block of code to determine the print label will be at risk of diverging from the block in PlaceRow. Even though they aren't returning the exact same label, there's some shared logic that I have a feeling we would want to co-locate:

https://github.com/opentripplanner/otp-react-redux/blob/bd9e51c915b5ec73895ee85506ceaf5cd4812c19/lib/components/narrative/line-itin/place-row.js#L170-L206

Case in point: I think the only reason this pull requests exists is because there is a significant split between the print and narrative itinerary components.

File: lib/util/itinerary.js [hunk 4]
```diff
@@ -402,6 +402,10 @@ export function getLegIcon (leg, customIcons) {
   return getIcon(iconStr, customIcons)
 }
 
+/**
+ * Get the configured company object for the given network string if the company
+ * has been defined in the provided companies array config.
+ */
 export function getCompanyForNetwork (networkString, companies = []) {
```
Comment (landonreed) [inline lib/util/itinerary.js]: I'm pretty sure this method can be private now.

**Comment by landonreed** [comment]
Also, I tried to run this code and ran into an error when I went to [this URL](http://localhost:8080/#/print/?ui_activeSearch=96gwy3fc9&ui_activeItinerary=0&fromPlace=1752%20SE%20Ladd%20Ave%2C%20Portland%2C%20OR%2C%20USA%2097214%3A%3A45.51043713898763%2C-122.65101946419323&toPlace=4849%20NE%20Mason%20St%2C%20Portland%2C%20OR%2C%20USA%2097218%3A%3A45.55300607630593%2C-122.61256722355365&date=2019-08-09&time=17%3A07&arriveBy=false&mode=BUS%2CTRAM%2CRAIL%2CGONDOLA%2CCAR_RENT%2CWALK&showIntermediateStops=true&optimize=QUICK&ignoreRealtimeUpdates=true&companies=CAR2GO&minTransitDistance=50%25&searchTimeout=10000&onlyTransitTrips=true). I think `customIcons` is not being passed into the `PrintableItinerary` component.

```
error      TypeError: Cannot read property 'customIconForLeg' of undefined
    at getLegIcon (itinerary.js:650)
    at AccessLeg.render (printable-itinerary.js:213)
    at eval (ReactCompositeComponent.js:773)
    at measureLifeCyclePerf (ReactCompositeComponent.js:73)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:772)
    at ReactCompositeComponentWrapper._renderValidatedComponent (ReactCompositeComponent.js:797)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:351)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
    at ReactDOMComponent.mountChildren (ReactMultiChild.js:234)
```

File: lib/components/narrative/printable/printable-itinerary.js (hunk 3 above)
Comment (evansiroky) [reply lib/components/narrative/printable/printable-itinerary.js]: I noticed that, but it has already been diverging due to different styling and the text requested from Myleen is diverging, so I opted to make things different. To begin with, the line-itin way of displaying things shows the itinerary as a graph with transfer points looking like nodes and having the header be the title of the node. The text in the RentedVehicleLeg does somewhat closely match, but should it be exactly the same? The print way of displaying things is more of a list style with the title of each thing being the overall instructions.

**Comment by evansiroky** [comment]
> Also, I tried to run this code and ran into an error when I went to [this URL](http://localhost:8080/#/print/?ui_activeSearch=96gwy3fc9&ui_activeItinerary=0&fromPlace=1752%20SE%20Ladd%20Ave%2C%20Portland%2C%20OR%2C%20USA%2097214%3A%3A45.51043713898763%2C-122.65101946419323&toPlace=4849%20NE%20Mason%20St%2C%20Portland%2C%20OR%2C%20USA%2097218%3A%3A45.55300607630593%2C-122.61256722355365&date=2019-08-09&time=17%3A07&arriveBy=false&mode=BUS%2CTRAM%2CRAIL%2CGONDOLA%2CCAR_RENT%2CWALK&showIntermediateStops=true&optimize=QUICK&ignoreRealtimeUpdates=true&companies=CAR2GO&minTransitDistance=50%25&searchTimeout=10000&onlyTransitTrips=true). I think `customIcons` is not being passed into the `PrintableItinerary` component. Is there a corresponding change needed in `trimet-mod-otp`?
> 
> ```
> error      TypeError: Cannot read property 'customIconForLeg' of undefined
>     at getLegIcon (itinerary.js:650)
>     at AccessLeg.render (printable-itinerary.js:213)
>     at eval (ReactCompositeComponent.js:773)
>     at measureLifeCyclePerf (ReactCompositeComponent.js:73)
>     at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:772)
>     at ReactCompositeComponentWrapper._renderValidatedComponent (ReactCompositeComponent.js:797)
>     at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:351)
>     at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:255)
>     at Object.mountComponent (ReactReconciler.js:43)
>     at ReactDOMComponent.mountChildren (ReactMultiChild.js:234)
> ```

I added something that should account for the posibility of the customIcons not being defined.

File: lib/components/narrative/printable/printable-itinerary.js (hunk 3 above)
Comment (landonreed) [reply lib/components/narrative/printable/printable-itinerary.js]: I'm not suggesting that the text should match between the two, but rather that the logic that processes the leg and delivers the text for the component should be co-located in a single method. It's almost more of a code management strategy to improve the chances that when modifications are made to the line-itin component we also remember to make the some modifications to the print view.

**Comment by landonreed** [comment]
Since I created the PR, I cannot approve it through the GH workflow, but this comment marks my stamp of approval.
